Realm/RLMSyncSession.h

```diff
@@ -233,4 +233,46 @@ NS_REFINED_FOR_SWIFT;
 
 @end
 
+/**
+ A task object which can be used to observe or cancel an async open.
+
+ When a synchronized Realm is opened asynchronously, the latest state of the
+ Realm is downloaded from the server before the completion callback is invoked.
+ This task object can be used to observe the state of the download or to cancel
+ it. This should be used instead of trying to observe the download via the sync
+ session as the sync session itself is created asynchronously, and may not exist
```

Comment:
Is this Cocoa specific? The sync session is created here: https://github.com/realm/realm-object-store/blob/master/src/impl/realm_coordinator.cpp#L311 Which is called synchronously as part of calling `RealmCoordinator::get_synchronized_realm()`

Reply:
The very first thing that Cocoa's asyncOpen does is dispatch to a background queue, and get_synchronized_realm() is called from there.